tests/tests/src/test_ramification.mjs

```diff
@@ -33,11 +33,12 @@ function f(x) {
 function f2(x) {
   let v = 0;
   let y;
-  v = 1;
   if (x.TAG === "A") {
+    v = 1;
```

Comment:
This is the only case where the optimisation was kicking.
It's pretty unlikely to fire to begin with, and when it fires the cases where one can legitimately move the common instruction up are pretty rare.
In this case, it would be OK because assigning to `v` does not change the value of `x.TAG`, because the two values cannot be aliases.